src/components/SocialMediaLinks/SocialMediaLinks.tsx

```diff
@@ -0,0 +1,195 @@
+  /** Optional layout */
+  layout?: LayoutTypes;
+  /** Optional array of social media platform types, urls, and label texts */
+  linksData?: SocialMediaLinkDataProps[];
+  /** Optional true/false to display names of platform along with icon */
+  showLabels?: boolean;
+  /** Optional size: medium, large, xlarge */
+  size?: SizeType;
+}
+
+/* Accepts an array containing one or more data objects, each representing a social media platform and optional overrides for the label and/or URL.
+ *
+ *  @returns an array of data objects for each requested platform type that includes type, iconName, labelText and url where the latter two may include values supplied to override the defaults.
+ */
+function getLinksData(platforms: SocialMediaLinkDataProps[]) {
+  let allData = [];
+  platforms.forEach((myPlatform) => {
+    // Test if the type is allowed
+    if (!socialMediaLinkTypeArray.includes(myPlatform.type)) {
+      console.error(
+        "NYPL Reservoir SocialMediaLinks: " +
+          myPlatform.type +
+          " is either not a supported platform or is misspelled.\n" +
+          "Supported platforms include: " +
+          socialMediaLinkTypeArray.join(", ")
+      );
+      return;
+    }
+
+    // Get the dataset for this platform.
+    let thisPlatformArray = socialMediaDataMap.filter(
+      (socialMediaSlice) => socialMediaSlice.type === myPlatform.type
+    );
+
+    // The filter function returns an array of objects, so let's just get the objects out of the array.
+    let thisPlatformData = thisPlatformArray[0];
+
+    // If a url value exists in myPlatform use it, else use the default.
+    let newUrl = myPlatform.url ?? thisPlatformData.url;
+
+    // If a labelText value exists in myPlatform use it, else use the default.
+    let newLabelText = myPlatform.labelText ?? thisPlatformData.labelText;
+
+    let thisObj = {
+      type: thisPlatformData.type,
+      iconName: thisPlatformData.iconName,
+      labelText: newLabelText,
+      url: newUrl,
+    };
+    allData.push(thisObj);
+  }); // end foreach
+
+  return allData;
+}
+
+/**
+ * The SocialMediaLinks component renders a list of links for accessing social media sites.
+ */
+export const SocialMediaLinks = chakra(
+  forwardRef<HTMLDivElement, SocialMediaLinksProps>((props) => {
+    const {
+      borders = "none",
+      color = "textDefault",
+      className,
+      id,
+      linksData,
+      layout = "row",
+      showLabels = false,
+      size = "medium",
+      ...rest
+    } = props;
+
+    // Turns out you can pass whatever props you want to this thing in order to do logic in the theme.
+    const styles = useMultiStyleConfig("SocialMediaLinks", {
+      variant: borders,
+      size: size,
+      color: color,
+      layout: layout,
+    });
+
+    if (showLabels && borders === "circular") {
+      console.error(
+        "NYPL Reservoir SocialMediaLinks: `showLabels` is set to true, but labels are disallowed when also using borders='circular'."
```

Comment:
The spec doc has a slightly different message: "NYPL Reservoir SocialMediaLinks: `showLabels` is set to true, but labels can not be shown with a circular border."

Reply:
I rewrote the error message for greater clarity.